ROVU/Core/src/robot/A_Star.java

```diff
@@ -93,23 +80,26 @@ private static Node findRoute(){
 
 	}
 	
-	
-	private static void neighborFringe (Node node){
+	/**
+	 * 
+	 * @param node
+	 */
+	private  void neighborFringe (Node node){
 		int i=0;
-		tk = new TreeMap<Double, Node>();
-		System.out.println("Null "+node.roomID);
+
+		System.out.println("Null "+node.nodeID);
 		for (Node e:node.neighbours){
 			if (!listContains(e,closed) && !e.wall){
-				Node temp = new Node(false, false, false, e.roomID, e.point);
+				Node temp = new Node(false, false, false, e.nodeID, e.point);
 				temp.parent=node;
```

Comment:
I see that you're accessing the variables of the node `temp` now, but with the changes applied in `Node`, you would need to use `temp.setParent(node)`. So please change the way the values have been assigned here.